examples/weather/pangu_weather/train_pangu_era5.py

```diff
@@ -203,122 +262,197 @@ def main(cfg: DictConfig) -> None:
+            num_samples_per_year=cfg.val.num_samples_per_year,
+            batch_size=cfg.val.batch_size,
+            device="cpu",
+            num_workers=cfg.val.num_workers,
+            shuffle=False,
+            use_cos_zenith=cfg.train.use_cosine_zenith,
+            cos_zenith_args={
+                "dt": 6.0,
+                "start_year": 1980,
+                "latlon_bounds": ((90, -90), (0, 360)),
+            },
+            latlon_resolution=img_size,
+        )
+        logger.success(
+            f"Rank {dist.rank}: Loaded validaton datapipe of size {len(validation_datapipe)}"
+        )
 
     @StaticCaptureEvaluateNoGrad(model=pangu_model, logger=logger, use_graphs=False)
-    def eval_step_forward(my_model, invar_surface, surface_mask, invar_upper_air):
-        invar = my_model.prepare_input(invar_surface, surface_mask, invar_upper_air)
-        return my_model(invar)
+    def eval_step_forward(my_model, invar, cos_zenith, surface_mask, outvar, weights):
+        # Multi-step prediction
+        invar = torch.concat([surface_mask, cos_zenith, invar], dim=1)
+        outpred = my_model(invar)
+        loss = torch.sum(
+            weights * (outpred - outvar) ** 2, dim=(0, -2, -1)
+        ) / torch.sum(weights)
+        return outpred, loss
 
     @StaticCaptureTraining(
-        model=pangu_model, optim=optimizer, logger=logger, use_graphs=False
+        model=pangu_model,
+        optim=optimizer,
+        logger=logger,
+        use_graphs=cfg.train.enable_graphs,
+        use_amp=cfg.train.enable_amp,
+        gradient_clip_norm=cfg.train.get("gradient_clip_norm", None),
     )
-    def train_step_forward(
-        my_model,
-        invar_surface,
-        surface_mask,
-        invar_upper_air,
-        outvar_surface,
-        outvar_upper_air,
-    ):
+    def train_step_forward(my_model, invar, cos_zenith, surface_mask, outvar, weights):
         # Multi-step prediction
         loss = 0
-        # Multi-step not supported
-        for t in range(outvar_surface.shape[1]):
-            invar = my_model.prepare_input(invar_surface, surface_mask, invar_upper_air)
-            outpred_surface, outpred_upper_air = my_model(invar)
-            invar_surface = outpred_surface
-            invar_upper_air = outpred_upper_air
-            loss += loss_func(outpred_surface, outvar_surface[:, t]) * 0.25 + loss_func(
-                outpred_upper_air, outvar_upper_air[:, t]
-            )
+        batch_size = outvar.shape[0]
+        for b in range(batch_size):
+            invar_ = invar[b : b + 1]
+            cos_zenith_ = cos_zenith[b : b + 1]
+            for t in range(outvar.shape[1]):
+                invar_ = torch.concat(
+                    [surface_mask, cos_zenith_[:, t : t + 1], invar_], dim=1
+                )
+                outpred = my_model(invar_)
+                loss += loss_func(outpred, outvar[b : b + 1, t], weights) / batch_size
+                invar_ = outpred
+
```

Comment:
Is there a reason this inner multistep loop cannot be run with batch_size > 1? Would be nice to support batched rollout training if it could fit in memory